tests/unit/conftest.py

```diff
@@ -10,62 +10,113 @@
+        tmp_path: A unique temporary directory path provided by pytest for test isolation.
+
+    Returns:
+        A callable that prepares the test environment.
+    """
+
+    def _prepare_test_env() -> None:
+        # Set the environment variable for the local storage directory to the temporary path.
         monkeypatch.setenv('CRAWLEE_STORAGE_DIR', str(tmp_path))
 
-        # Reset services in crawlee.service_container
-        cast(dict, service_container._services).clear()
+        # Initialize services in the service container with default values.
+        service_locator.set_configuration(Configuration())
+        service_locator.set_storage_client(MemoryStorageClient())
+        service_locator.set_event_manager(LocalEventManager())
+
+        # Reset the global state flags in the service locator.
+        service_locator._service_locator._configuration_was_set = False
+        service_locator._service_locator._storage_client_was_set = False
+        service_locator._service_locator._event_manager_was_set = False
 
-        # Clear creation-related caches to ensure no state is carried over between tests
+        # Clear creation-related caches to ensure no state is carried over between tests.
         monkeypatch.setattr(_creation_management, '_cache_dataset_by_id', {})
         monkeypatch.setattr(_creation_management, '_cache_dataset_by_name', {})
         monkeypatch.setattr(_creation_management, '_cache_kvs_by_id', {})
         monkeypatch.setattr(_creation_management, '_cache_kvs_by_name', {})
         monkeypatch.setattr(_creation_management, '_cache_rq_by_id', {})
         monkeypatch.setattr(_creation_management, '_cache_rq_by_name', {})
 
-        # Verify that the environment variable is set correctly
+        # Verify that the test environment was set up correctly.
         assert os.environ.get('CRAWLEE_STORAGE_DIR') == str(tmp_path)
+        assert service_locator._service_locator.configuration_was_set is False
+        assert service_locator._service_locator.storage_client_was_set is False
+        assert service_locator._service_locator.event_manager_was_set is False
 
-    return reset
+    return _prepare_test_env
+
+
+@pytest.fixture
+def cleanup_test_env() -> Callable[[], None]:
+    """Clean up and reset global state after a test has completed.
+
+    This fixture ensures that any modifications to the global state during a test are undone
+    after the test finishes. Restoring the environment for the subsequent tests.
+
+    Returns:
+        A callable that cleans up the test environment.
+    """
+
+    def _cleanup_test_env() -> None:
+        # Reset the flags in the service locator to indicate no services are explicitly set.
+        service_locator._service_locator._configuration_was_set = False
+        service_locator._service_locator._storage_client_was_set = False
+        service_locator._service_locator._event_manager_was_set = False
+
+    return _cleanup_test_env
```

Comment:
Can't we just inline this in `_isolate_test_environment`? Or do we use this one on its own somewhere?

Reply:
Currently, we only use `prepare_test_env` in some tests. However, this approach is consistent, and it's possible that we might need to use `cleanup_test_env` in the future as well. I don't see any issue in the current state.

Reply:
Well, it's one more fixture to maintain and I'm just not very sure about it being useful on its own.

Reply:
Ok, let's keep only `cleanup_test_env` fixture.